### PR TITLE
feat: added authentication functionality

### PR DESCRIPTION
**Pull request by md-amir-gauhar**
Screenshot (215)

### CODE REVIEW

File: src/backend/db/users.js [hunk 1]
```diff
@@ -9,10 +9,9 @@ import { formatDate } from "../utils/authUtils";
 export const users = [
   {
     _id: uuid(),
-    firstName: "Adarsh",
-    lastName: "Balika",
-    email: "adarshbalika@gmail.com",
-    password: "adarshBalika123",
+    username: "Roronoa Zoro",
```
Comment (Hamza-777) [inline src/backend/db/users.js:12]: Better if you keep test login as something simple like guest.

**Review by Hamza-777 — APPROVED** [review]
Looks good to merge

File: src/backend/db/users.js (hunk 1 above)
Comment (md-amir-gauhar) [reply src/backend/db/users.js:12]: Ok🥲

**Review by HrishikeshMahalle — APPROVED** [review]
Good code🔥

**Pull request merged**